### ISSUE TITLE
🐞 Typescript SDK - Intermittent Issues "Encountered an unknown error while requesting data via graphql"

### ISSUE THREAD

**Issue by levlaz** [dagger/dagger#8524]
### What is the issue?

When running a build with the Tyepscript SDK this error causes the entire build to fail. 

We have seen this happen in several different Typescript projects since 0.13 came out. 

### Dagger version

dagger v0.13.2 (registry.dagger.io/engine:v0.13.2) darwin/arm64

### Steps to reproduce

I run into this intermittently on this project: https://github.com/levlaz/medplum on the daggerize branch 

`dagger call build-matrix entries`

I am not able to reproduce this consistently 

### Log output

Error: Encountered an unknown error while requesting data via graphql

**Comment by jedevc** [comment]
Okay, so it *does* look like the pipeline is failing - the `npm ci` step fails: https://dagger.cloud/jedevc/traces/714d4614707c58ab16f8088da5ad1b48?span=cb510c70337818de#cb303a9ea768c3d2

![image](https://github.com/user-attachments/assets/4efa6f3b-520b-4af7-8315-3881e0160285)

However, what I don't understand is why this error message throws "Encountered an unknown error while requesting data via graphql" - this should be an `EXEC_ERROR` from the condition in https://github.com/dagger/dagger/blob/5ec03a604d50b721aaa65ba7c4db8c875696ac72/sdk/typescript/api/utils.ts#L206-L239

cc @TomChv - could the graphql library changes we made recently be causing this?

**Comment by jedevc** [comment]
I modified the exception to the following:

```diff
    // Just throw the unknown error
+   throw e;
    throw new UnknownDaggerError(
      "Encountered an unknown error while requesting data via graphql",
      {
        cause: e,
      },
    )
```

And get this error:

```
Error: response from query: input: medplum.buildMatrix resolve: call function "buildMatrix": process "tsx --no-deprecation --tsconfig /src/.dagger/tsconfig.json /src/.dagger/src/__dagger.entrypoint.ts" did not complete successfully: exit code: 1

Stderr:
Error: fetch failed
```

`fetch failed` doesn't seem like the right error message to be presenting here, not quite sure why we're not getting a graphql error?

**Comment by TomChv** [comment]
Because we catch the traces in the execution to not display it because it was too verbose for the user. But I think I'll rollback on that decision since it creates confusion.

`npm ci` isn't supposed to fail, and isn't related to that `graphql-request` error.
It's mostyl a engine connection error, I'll see in the graphql-request options if there's something we can do about that.

**Comment by jedevc** [comment]
If we're getting `context cancelled`, like in the traces above, is it possible that the graphql request is getting terminated before completion? It looks like it never completes from the trace/

**Comment by TomChv** [comment]
> If we're getting context cancelled, like in the traces above, is it possible that the graphql request is getting terminated before completion? It looks like it never completes from the trace.

Yeah that's probably that, I'll push a fix to get more information on the error.

**Comment by TomChv** [comment]
This should add more context to the actual error: https://github.com/dagger/dagger/pull/8544

**Comment by levlaz** [comment]
@jedevc I believe the pipeline step (i.e `npm ci`) is failing because of this error, not the other way around. 

The reason I say this is because I can hit this like 30% of the time and the rest of the the everything is fine. 

the other thing setting me off is that this issue appeared in many different projects all of a sudden starting in 0.13

**Comment by TomChv** [comment]
Might be fixed by #8576 based on https://discord.com/channels/707636530424053791/1288358859190308874/1293148162906914877
